pyscraper/wa/parse.py

```diff
@@ -1,186 +1,370 @@
+                    "against": item.VotesTotalAgainst.string,
+                    "abstain": item.VotesTotalAbstain.string,
+                    #"result_en": item.Vote_Result_Welsh.string,
+                    #"result_cy": item.Vote_Result_English.string,
+                    #"meeting_type": item.Meeting_type.string,
+                    "votes": {},
+                }
+            vote = str(item.Results_Result.string).lower()
+            if not int(item.Member_Id.string): continue
+            name = str(item.Member_name_English.string)
             try:
-                agenda_id = item.contribution_type
-                print(agenda_id)
-            except:
-                pass
-            continue
-            #['Agenda_Item_ID']
-            agenda_id = self.getTagValue(item, 'Agenda_Item_ID')
-            speaker_id = self.getTagValue(item, 'Member_Id')
-            ctype = self.getTagValue(item, 'contribution_type')
-            if current_speech == 0:
-                current_speech = self.new_speech(item)
-
-            if agenda_id != current_agenda_id:
-                self.display_speech(current_speech)
-                current_speech = self.new_speech(item)
-                current_speaker = speaker_id
+                person = memberList.match_by_id(item.Member_Id.string, self.date)
+            except KeyError:
+                raise Exception(f"Could not find person for {name} {item.Member_Id.string}")
+            self.divisions[vote_id]['votes'][person['id']] = (name, vote)
+
+    def parse_qnr(self, qnr):
+        soup = BeautifulSoup(qnr, 'xml')
+        current_agenda_en = ""
+        major = False
+        c = 0  # There are duplicate order IDs in the QNR
+        for item in soup.find_all(['XML_Plenary_QNR_Bilingual', 'XML_Plenary-FifthSenedd_QNR_Bilingual']):
+            if not major:
+                major = True
+                self.add_speech(item, MajorHeading, en='QNR', cy='QNR', suffix='mh')
+
+            agenda_en = str(item.Agenda_item_english.string)
+            if agenda_en != current_agenda_en and agenda_en:
+                current_agenda_en = agenda_en
+                self.display_heading_agenda(item, MinorHeading)
+
+            ctype = str(item.contribution_type.string)
+            if ctype == 'QNR':
+                self.display_speech(item, html=False, order_id=c)
+            elif ctype == 'ANR':
+                self.display_speech(item, double_escaped=True, order_id=c)
+            else:
+                raise Exception(f"Unknown contribution type {ctype}")
+            c += 1
+
+    def parse_plenary(self, text):
+        soup = BeautifulSoup(text, 'xml')
+
+        self.speeches = []
+        current_agenda_id = '0'
+        for item in soup.find_all(['XML_Plenary_Bilingual', 'XML_Plenary-FifthSenedd_Bilingual']):
+            #entry_id = item.Contribution_ID
+            #tv_spoken = item.contribution_spoken_seneddTv
+            #tv_translated = item.contribution_translated_seneddTv
```

Comment:
I assume these are not needed?

Reply:
Yeah, I left them in just as documentation on what was available in the XML, I guess.